File: java/src/main/java/com/sasteval/vuln/hard/CommandBuilder.java
package com.sasteval.vuln.hard;

/**
 * CWE-78: OS Command Injection (Cross-File Helper - Command Builder)
 * Part of a 3-file taint chain: CommandInjectionCrossFile -> CommandBuilder -> CommandRunner.
 */
public class CommandBuilder {

    public String buildPingCommand(String target) {
        // VULN: User input concatenated into shell command string
        return "ping -c 1 " + target;
    }
}

File: dotnet/Vuln/Hard/CommandBuilder.cs
namespace SastEval.Vuln.Hard;

/// <summary>
/// Helper class for CommandInjectionCrossFile - builds shell command with string concatenation.
/// </summary>
public static class CommandBuilder
{
    public static string BuildPingCommand(string target)
    {
        // VULNERABLE: User input concatenated into shell command
        return "ping -c 1 " + target;
    }
}
